File: src/transformers/features/protobuf/ProtobufCreateAssertDecodeTransformer.ts
import { ProtobufAssertDecodeProgrammer } from "../../../programmers/protobuf/ProtobufAssertDecodeProgrammer";

import type { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace ProtobufCreateAssertDecodeTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.factory({
      ...props,
      method: "protobuf.createAssertDecode",
      write: ProtobufAssertDecodeProgrammer.write,
    });
}

File: packages/typia/native/transform/features/protobuf/ProtobufCreateAssertDecodeTransformer.go
package protobuf

import (
  shimast "github.com/microsoft/typescript-go/shim/ast"
  nativecontext "github.com/samchon/typia/packages/typia/native/core/context"
  nativeprotobufprogrammers "github.com/samchon/typia/packages/typia/native/core/programmers/protobuf"
  nativeinternal "github.com/samchon/typia/packages/typia/native/transform/internal"
)

type protobufCreateAssertDecodeTransformerNamespace struct{}

var ProtobufCreateAssertDecodeTransformer = protobufCreateAssertDecodeTransformerNamespace{}

func (protobufCreateAssertDecodeTransformerNamespace) Transform(props nativeinternal.ITransformProps) *shimast.Node {
  return nativeinternal.GenericTransformer.Factory(nativeinternal.GenericTransformer_IProps{
    ITransformProps: props,
    Method:          "protobuf.createAssertDecode",
    Write: func(x nativecontext.IProgrammerProps) *shimast.Node {
      return nativeprotobufprogrammers.ProtobufAssertDecodeProgrammer.Write(nativeprotobufprogrammers.ProtobufAssertDecodeProgrammer_IProps(x))
    },
  })
}
